saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -364,6 +426,81 @@ def trigger_webhooks_async(
     )
 
 
+@app.task(
+    queue=settings.WEBHOOK_CELERY_QUEUE_NAME,
+    bind=True,
+)
+def generate_deferred_payloads(
+    self,
+    event_delivery_ids: list,
+    deferred_payload_data: dict,
+    queue: Optional[str] = None,
+):
+    deliveries = list(get_multiple_deliveries_for_webhooks(event_delivery_ids).values())
+    args_obj = DeferredPayloadData(**deferred_payload_data)
+    requestor = None
+    if args_obj.requestor_object_id and args_obj.requestor_model_name in (
+        RequestorModelName.APP,
+        RequestorModelName.USER,
+    ):
+        model = apps.get_model(args_obj.requestor_model_name)
+        requestor = model.objects.filter(pk=args_obj.requestor_object_id).first()
+
+    subscribable_object = (
+        apps.get_model(args_obj.model_name)
+        .objects.filter(pk=args_obj.object_id)
+        .first()
+    )
+    if not subscribable_object:
+        EventDelivery.objects.filter(pk__in=event_delivery_ids).update(
+            status=EventDeliveryStatus.FAILED
+        )
+        return
+
+    for delivery in deliveries:
+        event_type = delivery.event_type
+        webhook = delivery.webhook
+        request = initialize_request(
+            requestor,
+            event_type in WebhookEventSyncType.ALL,
+            event_type=event_type,
+            allow_replica=True,
+            request_time=args_obj.request_time,
+        )
+        data_promise = generate_payload_promise_from_subscription(
+            event_type=event_type,
+            subscribable_object=subscribable_object,
+            subscription_query=webhook.subscription_query,  # type: ignore
+            request=request,
+            app=webhook.app,
+        )
+
+        if data_promise:
+            data = data_promise.get()
+            if data:
+                data_json = json.dumps({**data})
+                with allow_writer():
+                    event_payload = EventPayload.objects.create_with_payload_file(
```

Comment:
In each place, payload generation is wrapped in transaction.atomic(). Shouldn't we also do this here?
Also this means, that for each delivery in deliveries, we are calling 2 or 3 db queries. Can't we make it in bulk? For example, using: `bulk_create_with_payload_files` ?